datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -608,6 +609,8 @@ protected void maybeCommitOffsetsInternal(Consumer<?, ?> consumer, boolean force
 
   protected void commitWithRetries(Consumer<?, ?> consumer, Optional<Map<TopicPartition, OffsetAndMetadata>> offsets)
       throws DatastreamRuntimeException {
+    preCommitHook();
```

Comment:
If preCommitHook() threw exception, the commit will not go through. Your current use-case to have audit related drain in the preCommitHook() also adds a dependency that if there is any failure in audit, the commits will not go through. Is this really expected ? Is it possible to have the audit system throwing errors while the regular producers are working fine.

Reply:
I made it explicit in javadoc that the overriding class needs to take care of handling exceptions from pre-commit hook.